internal/cmd/operator_list_operands.go

```diff
@@ -0,0 +1,128 @@
+	"fmt"
+	"io"
+	"os"
+	"sort"
+	"strings"
+	"text/tabwriter"
+	"time"
+
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/util/duration"
+	"sigs.k8s.io/yaml"
+
+	"github.com/operator-framework/kubectl-operator/internal/cmd/internal/log"
+	"github.com/operator-framework/kubectl-operator/pkg/action"
+)
+
+func newOperatorListOperandsCmd(cfg *action.Configuration) *cobra.Command {
+	l := action.NewOperatorListOperands(cfg)
+	output := ""
+	validOutputs := []string{"json", "yaml"}
+
+	cmd := &cobra.Command{
+		Use:   "list-operands <operator>",
+		Short: "List operands of an installed operator",
+		Args:  cobra.ExactArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			l.PackageName = args[0]
+
+			writeOutput := func(io.Writer, *unstructured.UnstructuredList) error { panic("writeOutput was not set") }
+			switch output {
+			case "json":
+				writeOutput = writeJSON
+			case "yaml":
+				writeOutput = writeYAML
+			case "":
+				writeOutput = writeTable
+			default:
+				log.Fatalf("invalid value for flag output %q, expected one of %s", output, strings.Join(validOutputs, "|"))
+			}
+
+			operands, err := l.Run(cmd.Context())
+			if err != nil {
+				log.Fatalf("list operands: %v", err)
+			}
+
+			if len(operands.Items) == 0 {
+				log.Print("No resources found")
+				return
+			}
+
+			sort.Slice(operands.Items, func(i, j int) bool {
+				if operands.Items[i].GetAPIVersion() != operands.Items[j].GetAPIVersion() {
+					return operands.Items[i].GetAPIVersion() < operands.Items[j].GetAPIVersion()
+				}
+				if operands.Items[i].GetKind() != operands.Items[j].GetKind() {
+					return operands.Items[i].GetKind() < operands.Items[j].GetKind()
+				}
+				if operands.Items[i].GetNamespace() != operands.Items[j].GetNamespace() {
+					return operands.Items[i].GetNamespace() < operands.Items[j].GetNamespace()
+				}
+				return operands.Items[i].GetName() < operands.Items[j].GetName()
+			})
+			if err := writeOutput(os.Stdout, operands); err != nil {
+				log.Fatal(err)
+			}
+		},
+	}
+	cmd.Flags().StringVarP(&output, "output", "o", output, fmt.Sprintf("Output format. One of: %s", strings.Join(validOutputs, "|")))
+	return cmd
+}
+
+func writeTable(w io.Writer, operands *unstructured.UnstructuredList) error {
+	var buf bytes.Buffer
+	tw := tabwriter.NewWriter(&buf, 3, 4, 2, ' ', 0)
+	if _, err := fmt.Fprintf(tw, "APIVERSION\tKIND\tNAMESPACE\tNAME\tAGE\n"); err != nil {
+		return err
+	}
+	for _, o := range operands.Items {
+		age := time.Since(o.GetCreationTimestamp().Time)
+		if _, err := fmt.Fprintf(tw, "%s\t%s\t%s\t%s\t%s\n", o.GetAPIVersion(), o.GetKind(), o.GetNamespace(), o.GetName(), duration.HumanDuration(age)); err != nil {
+			return err
+		}
+	}
+	if err := tw.Flush(); err != nil {
+		return err
+	}
+	if _, err := w.Write(buf.Bytes()); err != nil {
+		return err
+	}
+	return nil
+}
+
+func writeJSON(w io.Writer, operands *unstructured.UnstructuredList) error {
```

Comment:
Does this JSON formatting leaving the output in a state where another tool like `kubectl` can process it? One question I have is whether `kubectl` can take a JSON list as input and delete every member. I was envisioning that the `-o json` option would enumerate every entry in the list as a separate row so that it can easily be piped to other commands, but if a list works, then that's great as well

Reply:
Yep! This list works and is how kubectl typically outputs multiple objects.